server/controllers/finance/accounts/index.js

```diff
@@ -208,22 +209,85 @@ function lookupAccount(id) {
       a.classe, a.is_asset, a.reference_id, a.is_brut_link, a.is_charge,
       a.number, a.label, a.parent, a.type_id, a.is_title, at.type
     FROM account AS a JOIN account_type AS at ON a.type_id = at.id
-    WHERE a.id = ?;
-  `;
+    `;
+
+  sql += id ? ' WHERE a.id = ? ORDER BY CAST(a.number AS CHAR(15)) ASC;' : ' ORDER BY CAST(a.number AS CHAR(15)) ASC;';
 
   return db.exec(sql, id)
     .then(function(rows) {
       if (rows.length === 0) {
         throw new NotFound(`Could not find account with id ${id}.`);
       }
 
-      return rows[0];
+      return id ? rows[0] : rows;
     });
 }
 
+/**
+ * @method document
+ * @description generate chart of account as a document
+ */
+function document(req, res, next) {
+  const reportUrl = './server/controllers/finance/reports/accounts.report.handlebars';
+
+  lookupAccount()
+  .then(processAccountDepth)
+  .then(accounts => {
+    return rm.build(req, accounts, reportUrl);
+  })
+  .spread((doc, header) => {
+    res.set(header).send(doc);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+ * @method processAccountDepth
+ * @description get the depth of an account
+ */
+function processAccountDepth(accounts) {
+
+  return accounts.map(acc => {
+    let depth = getDepth(acc, 0);
+    acc.depth = depth;
+    return acc;
+  });
+
+  // recursive parent
+  function getDepth(account, depth) {
```

Comment:
These functions are good, but I would move their definitions out of the `processAccountDepth()` function.  They will be recreated and garbage collected every time `processAccountDepth()` is called.